examples/cpp/simple_local/CMakeLists.txt

```diff
@@ -2,10 +2,98 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-file(GLOB SRC_FILES ./*.cpp)
-file(GLOB H_FILES ./*.h)
+cmake_minimum_required(VERSION 3.10)
 
-add_demo(NAME object_detection_sample
-    SOURCES ${SRC_FILES}
-    HEADERS ${H_FILES}
-    DEPENDENCIES monitors models pipelines)
+# Multi config generators such as Visual Studio ignore CMAKE_BUILD_TYPE. Multi config generators are configured with
+# CMAKE_CONFIGURATION_TYPES, but limiting options in it completely removes such build options
+get_property(GENERATOR_IS_MULTI_CONFIG_VAR GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(NOT GENERATOR_IS_MULTI_CONFIG_VAR AND NOT DEFINED CMAKE_BUILD_TYPE)
+    message(STATUS "CMAKE_BUILD_TYPE not defined, 'Release' will be used")
+    # Setting CMAKE_BUILD_TYPE as CACHE must go before project(). Otherwise project() sets its value and set() doesn't take an effect
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel ...")
+endif()
+
+project(Samples)
+
+if(WIN32)
+    if(NOT "${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        message(FATAL_ERROR "Only 64-bit supported on Windows")
+    endif()
+
+    add_definitions(-DNOMINMAX)
+endif()
+
+if(MSVC)
+    add_compile_options(/wd4251 /wd4275 /wd4267  # disable some warnings
+                        /W3  # Specify the level of warnings to be generated by the compiler
+                        /EHsc)  # Enable standard C++ stack unwinding, assume functions with extern "C" never throw
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "^GNU|(Apple)?Clang$")
+    add_compile_options(-Wall)
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm64.*|aarch64.*|AARCH64.*)")
+  set(AARCH64 ON)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm.*|ARM.*)")
+  set(ARM ON)
+endif()
+if(ARM AND NOT CMAKE_CROSSCOMPILING)
+    add_compile_options(-march=armv7-a+fp)
+endif()
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+include(CMakeParseArguments)
+
+# add_demo(NAME <target name>
+#     SOURCES <source files>
+#     [HEADERS <header files>]
+#     [INCLUDE_DIRECTORIES <include dir>]
+#     [OPENCV_VERSION_REQUIRED <X.Y.Z>]
+#     [DEPENDENCIES <dependencies>])
+macro(add_demo)
```

Comment:
Can we rename this marco to "add_example" to reflect that we don't have demos here now?